snakemake/deployment/docker.py

```diff
@@ -0,0 +1,119 @@
+__author__ = "Johannes Köster"
+__copyright__ = "Copyright 2015-2020, Johannes Köster"
+__email__ = "johannes.koester@uni-due.de"
+__license__ = "MIT"
+
+import os
+import re
+import requests
+
+
```

Comment:
This is the start of a DockerContainer class that I thought would provide a minimal wrapper to parse a container uri (e.g., get the registry, repository, and version) and then do enough query to the manifest endpoint to determine if it exists or not. This class can be used by the WrapperSpec to determine if a given uri exists (or not). If it doesn't exist, we skip over using a container for the given wrapper.